client/cli/src/client_proxy.rs

```diff
@@ -343,6 +347,53 @@ impl ClientProxy {
         }
     }
 
+    /// Register an account as validator candidate with ValidatorConfig
+    pub fn register_validator(
```

Comment:
wow, I might suggest at this point we generate new config file that represents the public facing config for a node. The following looks hard to maintain and scary...

Reply:
I could punt on this is if you gave a really good example in the document section above, but this isn't really reusable whereas being able to point to a config file is.